chief.py

```diff
@@ -1,31 +1,81 @@
 from protocol import Parallel
 from http.server import BaseHTTPRequestHandler
+from docker_utils import get_container, create_archive
 
-class ChiefHTTPHandler(BaseHTTPRequestHandler):
+class ChiefWebUi(BaseHTTPRequestHandler):
     def do_POST(self):
-        
-        def upload_result():
-            # to do
-            pass
 
-        endpoints = {"/upload/result":upload_result}
+        def upload_data():
+            container = get_container("Butcher")
+            container.exec_run("mkdir -p /app", workdir="/")
+            container.exec_run("mkdir -p /app/Data", workdir="/")
+            container.exec_run("mkdir -p /app/Data/fragment_cache", workdir="/")
```

Comment:
These can all be one call. The `-p` option does two things, create parent directories and doesn't throw an error when the directory(s) already exist.